ext/openssl/xp_ssl.c

```diff
@@ -948,15 +950,34 @@ static int php_openssl_set_local_cert(SSL_CTX *ctx, php_stream *stream) /* {{{ *
 					certfile);
 				return FAILURE;
 			}
+			ctx_set = 1;
+		/* val is still local_cert/certfile since GET_VER_OPT_STRING("local_cert", certfile) */
+		} else if ((cert = php_openssl_x509_from_str(Z_STR_P(val))) != NULL) {
+			if (SSL_CTX_use_certificate(ctx, cert) != 1) {
+				X509_free(cert);
+				php_error_docref(NULL, E_WARNING,
+					"Invalid local cert `%s'; Check your device",
+					certfile);
+				return FAILURE;
+			}
+			ctx_set = 1;
+		}
+		if (ctx_set) {
 			GET_VER_OPT_STRING("local_pk", private_key);
-
 			if (private_key) {
 				char resolved_path_buff_pk[MAXPATHLEN];
 				if (VCWD_REALPATH(private_key, resolved_path_buff_pk)) {
 					if (SSL_CTX_use_PrivateKey_file(ctx, resolved_path_buff_pk, SSL_FILETYPE_PEM) != 1) {
 						php_error_docref(NULL, E_WARNING, "Unable to set private key file `%s'", resolved_path_buff_pk);
 						return FAILURE;
 					}
+				} else if (GET_VER_OPT("local_pk")) /* fill val with local_pk if any */ {
+					EVP_PKEY *pkey = php_openssl_pkey_from_zval(val /* local_pk */, 0/* not public */, NULL, 0);
```

Comment:
pls add space between `0` and `/*`

Reply:
done https://github.com/php/php-src/pull/6860/commits/27d8f44af90e66dcdb423979a15dc91f4d7f83e4